### PR TITLE
Update release notes

### PR DESCRIPTION
**Pull request by anissa111**
Part of #153

### CODE REVIEW

**Comment by kafitzgerald** [comment]
Thanks for catching this!  I'll see if I can add in a bit more detail quickly here.

File: docs/release-notes.rst [hunk 1]
```diff
@@ -4,6 +4,12 @@
 Release Notes
 =============
 
+v2023.09.0 (September 7, 2023)
+------------------------------
+
+Bug fix in ``add_height_from_pressure_axis()`` and added example documentation.
```
Comment (kafitzgerald) [inline docs/release-notes.rst:10]:
```suggestion
Bug Fixes
^^^^^^^^^^^^
* Unit and inferred height fixes in ``add_height_from_pressure_axis()``  by `Katelyn FitzGerald`_ in (:pr:`152`)

Documentation
^^^^^^^^^^^^^
* Additions to the examples gallery for ``find_local_extrema()``, ``plot_contour_labels()``, and ``plot_extrema_labels()`` by `Julia Kent`_ in (:pr:`145`)
```

**Review by kafitzgerald — CHANGES_REQUESTED** [review]
While we're in here, the usage examples links are broken in the last documentation section of the release notes (bc single ticks instead of double).  If it's easy, maybe we just fix those too?

File: docs/release-notes.rst (hunk 1 above)
Comment (anissa111) [reply docs/release-notes.rst:10]: That sounds fine with me, but we'd still want a top-level plain text summary, right?

**Comment by jukent** [comment]
Thanks again for putting this together!

**Review by kafitzgerald — APPROVED** [review]

**Pull request merged**